lib/plug/crypto/key_generator.ex

```diff
@@ -36,12 +36,17 @@ defmodule Plug.Crypto.KeyGenerator do
     digest = Keyword.get(opts, :digest, :sha256)
     cache = Keyword.get(opts, :cache)
 
-    if length > @max_length do
-      raise ArgumentError, "length must be less than or equal to #{@max_length}"
-    else
-      with_cache(cache, {secret, salt, iterations, length, digest}, fn ->
-        generate(mac_fun(digest, secret), salt, iterations, length, 1, [], 0)
-      end)
+    cond do
+      length > @max_length ->
+        raise ArgumentError, "length must be less than or equal to #{@max_length}"
+
+      !is_integer(iterations) or iterations < 1 ->
```

Comment:
We use `!` when we are dealing with any value and want to test for truthiness/falsey-ness, and `or` when dealing with strictly boolean conditions.

```suggestion
      not is_integer(iterations) or iterations < 1 ->
```